docs/proposals/cluster-backup-integration.md

```diff
@@ -0,0 +1,104 @@
+
+- During the cluster creation process, an option will be provided to select a backup to restore. In this case, an initial restore request will be created as a step of the cluster provisioning process. 
+
+### UI/Rest API
+TBD
+
+### API
+
+The API will include any KKP integration parameters needed, along with references to any KKP specific objects (e.g. provider presets).
+
+Additionally, those objects will wrap velero native objects. This pattern gives the users the ability to customize the backup/restore/destination configuration further then exposed via the UI, while simplifying the the implementation process, object reconciliation  and reduced the effort to maintain the future updates from velero side. 
+
+
+The integration will provide the user with the following types:
+- **CLusterBackup**
+```yaml
+TBD
+```
+This type will wrap the `Schedule` velero type. Additionally, it will specify the source cluster from which the backup is created. th The status of this type will reflect the last executed backup status and the last successful backup status.
+
+- **ClusterRestore**
+```yaml
+TBD
+```
+This type will wrap the `Restore` velero type. Additionally, it will specify which cluster to apply the restore resources on. This will enable the cluster migration feature. The status for this object should reflect the current status of the velero restore running on the user cluster.
+
+- **CLusterBackupDestination**
+```yaml
+TBD
+```
+This type will wrap both the `BackupStorageLocation` and `VolumeSnapshotLocation` velero types. Additionally, it will reference presets and related provider configuration from KKP to include in the velero types.
+
+
+### Controller(s)
+
+#### ClusterBackup Controller
+The  ClusterBackup controller will be deployed on the seed cluster in the user cluster namespace. It will watch the ClusterBackup API types on the seed cluster and apply them to the user cluster API. 
+
+The objects will be applied in a top-down fashion. The ClusterBackup will make sure only the changes applied via the KKP API is kept on the user cluster. Changes to the velero CRs applied by the user on the cluster API side will be reverted to maintain consistency.
+
+Velero will be deployed on the seed cluster in the user cluster namespace. This is provides additional resilience to the setup, since the backups will not be affected by the state of the user cluster (e.g. limited capacity, issues with the nodes) 
+
+**Note:** We need to investigate the feasibility of using this with other providers. Technically it should work with all providers that support CSI snapshots, since we will be connecting to the provider API from the same plane as the kubernetes API. However, we should confirm this.
```

Comment:
OpenStack is the priority provider, but note here: we also need to make a priority list for supporting them.

Reply:
Your initial scope mentions that we will not support community maintained providers. Since Openstack _is_ a community provider, and is a priority, we will focus on getting the initial implementation based on it. Theoretically, all the velero supported providers should also work, but other community providers might need additional work and should be prioritized for future releases.

Reply:
Exactly, okay!